src/test/qgstest.h

```diff
@@ -292,13 +296,64 @@ class TEST_EXPORT QgsTest : public QObject
       return result;
     }
 
+    bool checkLongStr( const char *file, const char *function, int line, const QString &name, const QString &referenceName, const QByteArray &actualStr )
+    {
+      QString subPath = "control_images/" + mControlPathPrefix + "/expected_" + name + "/" + "expected_" + referenceName;
+      QString expectedPath = testDataPath( subPath );
+      QFile expectedFile( expectedPath );
+      if ( ! expectedFile.open( QFile::ReadOnly  | QIODevice::Text ) )
+        return false;
+      QByteArray expectedStr = expectedFile.readAll();
+
+      if ( actualStr.size() != expectedStr.size() )
+        qWarning() << "Array have not the same length (actual vs expected): " << actualStr.size() << "vs" << expectedStr.size();
+
+      for ( int i = 0; ; i += 100 )
+      {
+        QByteArray act = actualStr.mid( i, 100 );
+        QByteArray exp = expectedStr.mid( i, 100 );
+        if ( act != exp )
+        {
+          QString actualPath = QDir::tempPath() + "/actual_" + name + "_" + referenceName;
+          QFile actualFile( actualPath );
+          if ( actualFile.open( QFile::WriteOnly | QIODevice::Text ) )
+            actualFile.write( actualStr );
+          else
+            qWarning() << "Unable to write actual data to file" << actualPath;
```

Comment:
I prefer having { } even for single line, moreover when there is no new line after like here. But it's not defined into our style guide :(

Reply:
done!